docs/cloud/rahti/storage/objectstorage.md

```diff
@@ -0,0 +1,280 @@
+
+If you don't have `access_key_id` and `secret_access_key`, you need to source your Pouta project and then use this command to create credentials:  
+
+```sh
+openstack ec2 credentials create
+```
+
+Once created, create your `rclone.conf` file:
+
+```ini
+[default]
+type = s3
+provider = Other
+env_auth = false
+access_key_id = {ACCESS_KEY_ID}
+secret_access_key = {SECRET_ACCESS_KEY}
+endpoint = a3s.fi
+acl = private
+```
+
+_Replace `{ACCESS_KEY_ID}` and `{SECRET_ACCESS_KEY}` by your own credentilas._ 
+
+Create a `rclone.sh` script:  
+
+```sh
+#!/bin/sh -e
+
+rclone copy "/mnt/" "default:{BUCKET}"
+
+echo "Done!"
+```
+
+_Replace `{BUCKET}` by the target bucket where you want to backup your files._  
+
+Then, you have to create your own custom `rclone` Docker image:  
+
+```Dockerfile
+FROM rclone/rclone
+
+COPY rclone.conf /.rclone.conf
+COPY rclone.sh /usr/local/bin/
+RUN chmod +x /usr/local/bin/rclone.sh
+```
+
+Once all this done, you can deploy your `rclone` pod. You can use this example:  
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: rclone
+spec:
+  containers:
+  - name: copys3
+    image: <your_rclone_image>
+    command: ["/usr/local/bin/rclone.sh"]
+    resources:
+      limits:
+        memory: "128Mi"
+        cpu: "500m"
+    volumeMounts:
+    - name: vol-to-backup
+      mountPath: /mnt/
+  volumes:
+  - name: vol-to-backup
+    persistentVolumeClaim:
+      claimName: nginx-pvc # Must match the PVC name that you want to backup
+```
+
+!!! Warning  
+    If your `PersistentVolumeClaim` is `ReadWriteOnce`, you have to scale down the `nginx` deployment to let the pod running rclone mount the volume.  
+    Use this command to proceed: `oc scale --replicas=0 deploy/nginx`  
+    If your `PersistentVolumeClaim` is `ReadWriteMany`, there is no need to scale down your deployment.  
+    You can verify with this command: `oc get pvc`. You should see either `RWO` or `RWX`.
+
+The pod will run and backup the content of your PVC to Allas. Don't forget to scale up your origin deployment (`oc scale --replicas=1 deploy/nginx`)
+
+There are PROS and CONS with this solution:  
+- PROS: You run the pod in your Rahti project  
+- CONS: If your PVC is `ReadWriteOnce`, a downtime is necessary.  
+
+## Second example: using bash script
+
+For the following script to work, we assume that you have the `rclone` command-line program installed and Allas bucket name is created. The `rclone.conf` should be set on your local system like described above example. For example, `rclone.conf` path could be located in `~/.config/rclone/rclone.conf`. More information on creating [Allas bucket](../../../data/Allas/using_allas/rclone.md). This script will buckup an application deployed in Rahti and the application has the `volumeMounts` `mountPath` name `/backup` as an example.
+
+
+```bash
+#!/bin/env bash
+
+# Set your pod name, source directory, and destination directory
+POD_NAME="backup-volume" # Your pod name 
```

Comment:
The name of the pod will change with every redeploy.

Could this be set as a parameter?

Reply:
I was thinking of putting a command like:
```
$(oc get pods -o yaml | yq '.items[0].metadata.name')
```

But it will only take the first pod

Reply:
I parameterized it to be passed as an argument to the script.